src/python_testing/TC_MCORE_FS_1_2.py

```diff
@@ -0,0 +1,185 @@
+                 TestStep(3, "TH_FSA completes commissioning of TH_SED_DUT using the Enhanced Commissioning Method."),
+                 TestStep(4, "Commission TH_SED_L onto DUT_FSA’s fabric using the manufacturer specified mechanism."),
+                 TestStep(5, "TH_FSA waits for subscription report from a the Bridged Device Basic Information clusters provided by DUT_FSA to identify the presence of a Bridged Node endpoint with a UniqueID matching the UniqueID provided by the BasicInformationCluster of the TH_SED_L."),
+                 TestStep(6, "TH_FSA initiates commissions of TH_SED_L by sending the OpenCommissioningWindow command to the Administrator Commissioning Cluster on the endpoint with the uniqueID matching that of TH_SED_L."),
+                 TestStep(7, "TH_FSA completes commissioning of TH_SED_L using the Enhanced Commissioning Method.")]
+        return steps
+
+    @async_test_body
+    async def test_TC_MCORE_FS_1_2(self):
+        self.is_ci = self.check_pics('PICS_SDK_CI_ONLY')
+        min_report_interval_sec = self.user_params.get("min_report_interval_sec", 0)
+        max_report_interval_sec = self.user_params.get("max_report_interval_sec", 10 * 60)
+        timeout_delay_sec = self.user_params.get("timeout_delay_sec", max_report_interval_sec * 2)
+
+        self.step(1)
+
+        # Subscribe to the UniqueIDs
+        unique_id_queue = queue.Queue()
+        subscription_contents = [
+            (Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID)  # On all endpoints
+        ]
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=subscription_contents,
+            reportInterval=(min_report_interval_sec, max_report_interval_sec),
+            keepSubscriptions=False
+        )
+        attribute_handler = AttributeChangeAccumulator(
+            name=self.default_controller.name, expected_attribute=Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID, output=unique_id_queue)
+        sub.SetAttributeUpdateCallback(attribute_handler)
+        sub_handlers.append(attribute_handler)
+
+        logging.info("Waiting for First BridgedDeviceBasicInformation.")
+        start_time = time.time()
+        elapsed = 0
+        time_remaining = timeout_delay_sec
+
+        th_sed_dut_bdbi_endpoint = -1
+        th_sed_dut_unique_id = -1
+
+        while time_remaining > 0 and th_sed_dut_bdbi_endpoint < 0:
+            try:
+                item = unique_id_queue.get(block=True, timeout=time_remaining)
+                client_name, endpoint, attribute, value = item['name'], item['endpoint'], item['attribute'], item['value']
+
+                # Record arrival of an expected subscription change when seen
+                if attribute == Clusters.BridgedDeviceBasicInformation.Attributes.UniqueID:
```

Comment:
You shouldn't be getting back reports for other attributes - you only subscribed to one

I think you can eliminate all of this by just using the AttributeChangeCallback from support.